proxy/mux.go

```diff
@@ -45,13 +46,20 @@ type mux struct {
 
 	// Connection watcher
 	connTracker *connTracker
+
+	// OCSP staple cache and responder
+	stapler stapler.Stapler
+
+	// Unsubscribe from staple updates
+	stapleUpdatesC chan *stapler.StapleUpdated
+	stopStapleC    chan struct{}
 }
 
 func (m *mux) String() string {
 	return fmt.Sprintf("mux(%d, %v)", m.id, m.state)
 }
 
-func New(id int, o Options) (*mux, error) {
+func New(id int, st stapler.Stapler, o Options) (*mux, error) {
```

Comment:
yeah, but it's actually always used by the server, so if OCSP is turned on and the stapler is nil  that would be a configuration error, what I wanted to avoid by making it non-optional